core/git_command.py

```diff
@@ -124,7 +125,7 @@ def raise_error(msg):
                                  cwd=working_dir or self.repo_path,
                                  env=environ,
                                  startupinfo=startupinfo)
-            stdout, stderr = p.communicate(stdin.encode(encoding="UTF-8") if stdin else None)
+            stdout, stderr = p.communicate((stdin.encode(encoding="UTF-8") if encode else stdin) if stdin else None)
```

Comment:
I don't think we can rely on `i18n.filesencoding` since  one project can have multiple encodings without a problem in git. I think we have couple of this project ourself.
I will implement it to use the same encoding to stage a hunk as showing the diff.